lib/galaxy/schema/library_contents.py

```diff
@@ -0,0 +1,327 @@
+import json
+from enum import Enum
+from typing import (
+    Any,
+    Dict,
+    List,
+    Optional,
+    Union,
+)
+
+from pydantic import (
+    ConfigDict,
+    Field,
+    RootModel,
+)
+from pydantic.functional_validators import field_validator
+from typing_extensions import (
+    Annotated,
+    Literal,
+)
+
+from galaxy.schema.fields import (
+    DecodedDatabaseIdField,
+    EncodedDatabaseIdField,
+    EncodedLibraryFolderDatabaseIdField,
+    LibraryFolderDatabaseIdField,
+    ModelClassField,
+)
+from galaxy.schema.schema import (
+    Model,
+    TagCollection,
+)
+
+
+class UploadOption(str, Enum):
+    upload_file = "upload_file"
+    upload_paths = "upload_paths"
+    upload_directory = "upload_directory"
+
+
+class CreateType(str, Enum):
+    file = "file"
+    folder = "folder"
+    collection = "collection"
+
+
+class LinkDataOnly(str, Enum):
+    copy_files = "copy_files"
+    link_to_files = "link_to_files"
+
+
+class LibraryContentsCreatePayload(Model):
+    create_type: CreateType = Field(
+        ...,
+        description="the type of item to create",
+    )
+    upload_option: UploadOption = Field(
```

Comment:
Can you mark direct upload through this API endpoint as deprecated ? We should aim at using the fetch data API, which can create and upload files into libraries and folders.

Reply:
Is it ok to do it like this: https://github.com/galaxyproject/galaxy/pull/18838/commits/800e96fa5c78db82dcc4282f9c97637e71b7115a

Reply:
Yes, thank you!